source/common/common/utility.cc

```diff
@@ -67,6 +71,60 @@ bool StringUtil::atoul(const char* str, uint64_t& out, int base) {
   }
 }
 
+absl::string_view StringUtil::ltrim(absl::string_view source) {
+  const absl::string_view::size_type pos = source.find_first_not_of(WhitespaceChars);
+  if (pos != absl::string_view::npos) {
+    source.remove_prefix(pos);
+  } else {
+    source.remove_prefix(source.size());
+  }
+  return source;
+}
+
+absl::string_view StringUtil::rtrim(absl::string_view source) {
+  const absl::string_view::size_type pos = source.find_last_not_of(WhitespaceChars);
+  if (pos != absl::string_view::npos) {
+    source.remove_suffix(source.size() - pos - 1);
+  } else {
+    source.remove_suffix(source.size());
+  }
+  return source;
+}
+
+absl::string_view StringUtil::trim(absl::string_view source) { return ltrim(rtrim(source)); }
+
+bool StringUtil::findToken(absl::string_view source, absl::string_view delimiters,
+                           absl::string_view key_token, bool trim_whitespace) {
+  const std::vector<absl::string_view> tokens = splitToken(source, delimiters, trim_whitespace);
+  if (trim_whitespace) {
+    for (auto token : tokens) {
+      if (key_token == trim(token)) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
Now that you are passing trim_whitespace through, you can just delete lines 99-106

Reply:
If we do that, split strings will NOT be trimmed and findToken will produce a completely different result. 
```
auto s = StringUtil::splitToken("abc; type=text", ";=", "type", true);

s[0] == "abc", s[1] == " type", s[2] == " text" 
```
Note that s[1] would still need to be trimmed in order to have an exactly match of word `type`. As I understand, 
the only difference between the functors SkipEmpty() and SkipWhitespace(), is that SkipWhitespace returns false if a string contains ONLY whitespaces.

https://github.com/abseil/abseil-cpp/blob/be40fdf1a86f4956d2f8125a7b6bd6d34e133c2d/absl/strings/str_split.h#L331

I'm not sure that we really want that.

Reply:
OK that's fine then.  I'm confused though: https://github.com/abseil/abseil-cpp/blob/be40fdf1a86f4956d2f8125a7b6bd6d34e133c2d/absl/strings/str_split.h#L358 looks like it has the expected behavior, no?

Reply:
If you look at the example that you just gave me, v[0] has leading and trailing whitespaces, which means that in order to match it with 'a', both sides would have to be trimmed. Right?
``` 
v[0] == " a ", v[1] == "b"
```

Reply:
My apologies; you are absolutely correct.  I totally misread the str_split.h.

Reply:
No worries Joshua! Thanks for the awesome code review!